master/usage/ipv6.md

```diff
@@ -1,31 +1,158 @@
+
+To enable IPv6 operation of a {{site.prodname}} deployment follow the
+steps below.
+
+1. Download the Calico manifest with the calico-config ConfigMap and
+   calico-node Daemonset as `calico.yaml`.
+1. Update the `calico.yaml` file with the following changes:
+   - If the ipam section of the `cni_network_config` has
+     `"type": "calico-ipam"` then it should be modified to have
+     [IPv4 assigning disabled and IPv6 assigning
+     enabled](/{{page.version}}/reference/cni-plugin/configuration#ipam).
+   - Add the following environment variables to the calico-node Daemonset
+
+     ```
+     - name: CALICO_IPV6POOL_CIDR
+       value: "fd20::0/112"
+     - name: IP6
+       value: "autodetect"
+     ```
+
+     - Set the `CALICO_IPV6POOL_CIDR` environment variable to the desired pool,
+       it should match the `--cluster-cidr` passed to the
+       kube-controller-manager and to kube-proxy.
+   - Ensure the environment variable `FELIX_IPV6SUPPORT` is set `true`
+     on the calico-node Daemonset.
+1. Apply the `calico.yaml` manifest.
+
+### DNS changes
+
+It will probably be necessary to modify your DNS pod for IPv6. If you are using
+[kube-dns](/{{page.version}}/getting-started/kubernetes/installation/manifests/kubedns.yaml), 
+then the following changes will ensure IPv6 operation.
+
+- Update the image versions to at least `1.14.8`.
+- Ensure the clusterIP for the DNS service matches the one specified to
+  the kubelet as `--cluster-dns`.
+- Add `--dns-bind-address=[::]` to the arguments for the kubedns container.
+- Add `--no-negcache` to the arguments for the dnsmasq container.
+- Switch the arguments on the sidecar container from
+  ```
+  --probe=kubedns,127.0.0.1:10053,kubernetes.default.svc.cluster.local,5,A
+  --probe=dnsmasq,127.0.0.1:53,kubernetes.default.svc.cluster.local,5,A
+  ```
+  to
+  ```
+  --probe=kubedns,127.0.0.1:10053,kubernetes.default.svc.cluster.local,5,SRV
+  --probe=dnsmasq,127.0.0.1:53,kubernetes.default.svc.cluster.local,5,SRV
+  ```
+
+## Enabling IPv6 with OpenStack
+
+### Prerequisites
 
 When using {{site.prodname}} with a VM platform (e.g. OpenStack), obtaining IPv6
 connectivity requires certain configuration in the guest VM image:
 
--   When it boots up, the VM should issue a DHCPv6 request for each of
-    its interfaces, so that it can learn the IPv6 addresses that
-    OpenStack has allocated for it.
--   The VM must be configured to accept Router Advertisements.
--   If it uses the widely deployed DHCP client from ISC, the VM must
-    have a fix or workaround for [this known
-    issue](https://kb.isc.org/article/AA-01141/31/How-to-workaround-IPv6-prefix-length-issues-with-ISC-DHCP-clients.html).
+-  When it boots up, the VM should issue a DHCPv6 request for each of
+   its interfaces, so that it can learn the IPv6 addresses that
+   OpenStack has allocated for it.
+-  The VM must be configured to accept Router Advertisements.
+-  If it uses the widely deployed DHCP client from ISC, the VM must
```

Comment:
You need to switch the it and the VM because the current sequence causes an unclear referent. "If it uses the widely deployed DHCP client from ISC, the VM " --> "If the VM uses the widely deployed DHCP client from ISC, it "